### ISSUE TITLE
Dynamic outputHashing option in angular.json (Regex || Function ) for enable/disable per file

### ISSUE THREAD

**Issue by DutchKevv** [angular/angular-cli#18283]
# 🚀 Feature request

### Command (mark with an `x`)

- [ ] new
- [x ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ x] config
- [ ] help
- [ ] version
- [ ] doc

### Description
With our use case, we need 1 file without outputHashing, all the other should become hashed. It doesn't seem to be possible now without make a custom afterbuild step / custom webpack config.

### Describe the solution you'd like
In the angular.json, in the 'outputHashing' option, allow a regex || function to test the files for enabled/disabled hashing.

### Describe alternatives you've considered
Custom webpack with a custom 'AfterEmitPlugin' hook, that reverts the given file to a non-hashed filename.

**Comment by alan-agius4** [comment]
@DutchKevv, what is the use-case of having a single file without output-hashing?

**Comment by DutchKevv** [comment]
@alan-agius4 

Well, maybe there is a better way over-all for this use case, but for now its like this:

We use Angular for our web app + admin app and use this same code-base also for our Cordova app. 

There are some pre-angular loading actions that need to happen, mainly for Cordova so we have created a 'core' typescript package that gets loaded _before_ any <script src="angular..."> in the index.html file, to start setting up some stuff and for optimization.

Within this Core.ts file, we use some interfaces etc from the Angular codebase... So ideally we wanted to have Angular watch for file changes, in both core directory and 'app' directory,  let Angular build the core.ts along with the 'app' as if its part of the Angular App, but to NOT be included in the Angular final builds (The Angular 'App' only uses some interfaces from Core, so its not included in the build). This core package is stand-alone and created as a separate output file / bundle.

So we are using custom.webpack.ts, added an entry file (core.ts) so its becomes a separate package, so the Angular compiler still watches and lints etc, and it works perfectly (y). https://github.com/angular/angular-cli/issues/17913
Only thing is that it gets a hashed output name, so we cannot pre-define it in our cordova-build / index.html without an external post-build step that removes the hash from the file name.



IN SHORT:

We use the Angular CLI to watch + compile a standalone Typescript module along side the Angular APP,  so that watchers / interfaces / linting / strong typing etc works as if it part of our Angular App, but we do not include it in our final build, but instead let webpack build a stand-alone module that we manually include in our Cordova / Web index.html. 

We could have a complete second webpack config, but that's a lot more work / error prone than the current setup where angular cli 'own' this package and builds it with the same tsconfig.json / configuration as our app does, making it also build an ES5 + ES2015 when Angular is configured like that, etc etc.

Hope is makes sense.

Thanks in advance

**Comment by alan-agius4** [comment]
Hi @DutchKevv, thanks for the detailed explanation.

This seems not to be a typical workflow for users using the Angular CLI where it will not benefit the majority of users will not benefit from this option.

Our recommandation would be to extend the Angular CLI using 3rd party tools such as https://github.com/manfredsteyer/ngx-build-plus or https://github.com/just-jeb/angular-builders

Thanks for your understanding.
